deployment/src/main/java/io/quarkiverse/helm/deployment/HelmProcessor.java

```diff
@@ -291,13 +299,11 @@ private void deleteOutputHelmFolderIfExists(Path outputFolder) {
         }
     }
 
-    private Path getInputDirectory(HelmChartConfig config, Project project) {
-        Path path = Paths.get(config.inputDirectory());
-        if (!path.isAbsolute()) {
-            return project.getRoot().resolve(path);
-        }
-
-        return path;
+    private Path getInputDirectory(HelmChartConfig config, Project project,
+            Optional<CustomKubernetesOutputDirBuildItem> customKubernetesOutputDir) {
+        return customKubernetesOutputDir
+                .map(CustomKubernetesOutputDirBuildItem::getOutputDir)
+                .orElse(project.getRoot().resolve(Paths.get(config.inputDirectory())));
```

Comment:
The input directory is where users can provide their custom resources for the Helm generation, so we can't use the configured output directory of the Kubernetes generation here.